mlir/test/IR/diagnostic-handler-metadata.mlir

```diff
@@ -0,0 +1,19 @@
+// RUN: mlir-opt %s -pass-pipeline="builtin.module(func.func(test-diagnostic-metadata))" -o - 2>&1 | FileCheck %s
+// This test verifies that diagnostic handler can filter the diagnostic whether to emit the errors.
+
+// CHECK-LABEL: Test 'test'
+// CHECK-NEXT: 8:3: error: test diagnostic metadata
+// CHECK-NOT: 13:3: error: test diagnostic metadata
```

Comment:
Instead of using FileCheck, can't we use the `--verify-diagnostic` mode of mlir-opt?

Reply:
Ah, yes, I was following [what `diagnostic-handler-filter.mlir` is done for checking the diagnostics](https://github.com/llvm/llvm-project/blob/2e57e6366677390110f5382894c8afeba8da7419/mlir/test/IR/diagnostic-handler-filter.mlir#L5-L6). 

`-verify-diagnostic` is a much better way, updated the test.